Library/Homebrew/cmd/update.sh

```diff
@@ -415,6 +415,17 @@ user account:
 EOS
   fi
 
+  # we need `gh` if the user enables attestation verification
+  if [[ -n "${HOMEBREW_VERIFY_ATTESTATIONS}" && ! -x "${HOMEBREW_GH}" ]]
```

Comment:
Think it's probably a bit early to have this in `update.sh`

Reply:
Agreed -- @Bo98 points out that `ensure_executable!` is the right approach here anyways, this is way too invasive for now 😅